src/qwed_new/core/stats_verifier.py

```
@@ -28,6 +32,54 @@
+    """Best-effort deterministic fingerprint of the input dataset.
+
+    Binds the verification outcome to the specific data that was analyzed so the
+    result can be replayed/audited against the same dataset. Uses pandas'
+    canonical per-row hasher (dtype/NaN aware) and hashes the digest bytes. This
+    is advisory evidence only — a hashing failure never changes the (already
+    fail-closed) verdict, so it degrades to ``None`` instead of raising.
+    """
+    try:
+        row_hashes = pd.util.hash_pandas_object(df, index=True)
+        return hashlib.sha256(row_hashes.values.tobytes()).hexdigest()
+    except Exception:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)